src/base/settingsstorage.cpp

```diff
@@ -34,18 +36,45 @@
 
 #include "logger.h"
 #include "utils/fs.h"
-#include "settingsstorage.h"
+
+#ifdef Q_OS_MAC
+#define QSETTINGS_SYNC_IS_SAVE //whether QSettings::sync() is "atomic"
+#endif
 
 namespace
 {
-    inline QSettings *createSettings(const QString &name)
-    {
+    // Incapsulates srialization of settings in "atomic" way.
+    // write() does not leave half-written files,
+    // read() has a workaround for a case of powerloss during a previous serialization
+    class TransactionalSettings {
```

Comment:
This is not your first PR. When you will accustom yourself to respect the coding style? Fix it everywhere! (and in other PRs too)